### ISSUE TITLE
LAST: display one a fixed number of changes

### ISSUE THREAD

**Issue by opn** [fedwiki/wiki-plugin-activity#28]
I suggest a new verb "LAST n" (when n is an integer) which would display the last n changes.

## Rational 
There are many occasions where I would like to display a minimal simple history of recent changes but cannot as the plugin is designed to render an arbitrary number of changes. This takes up an unpredictable amount of space (often too much space, or sometimes none)

For instance I may wish to place a short list of recent changes to the site on the **Welcome Visitors** page. SINCE 1 week may show hundreds of changes or none, where I would like to show just 5 changes regardless of when they occurred.

## Ugly Rendering
In addition the rendering of the results is unnecessarily complex / ugly for a minimal presentation:
```
Page Activity
  since Wed Aug 10 2016
  excluding neighborhood
```
For a minimal presentation we just need the title and a list of links. Possibly/optionally including the data of the change. With the following markup:
```
NEIGHBORHOOD no
NARRATIVE
LAST 3
```
Rendering something equivalent (in markdown) to:
```
# Recent changes to this site
*  [[Some Page]] - Aug 10 2016
*  [[Another Page]] - Aug 9 2016
*  [[Last Page]] - Jan 3 2016
```

**Comment by WardCunningham** [comment]
The plugin inserts separators that can be confusing when limits are placed on output. For example `Within a Month` is a helpful separator easily interpreted by even the calendar challenged reader such as myself. However it won't be correct if the Activity plugin is not allowed to show all of the activity within the month.

We've tried several versions of headings (and footings) to make clear when the common meaning of the words would suggest inclusiveness that is no longer present. If they are judged ugly, it might be because they require too much mental date calculation to interpret.

I suggest we abandon trying to explain how the plugin is configured and leave that to the page author who can add their own introductory paragraphs if they so choose. Let them suffer in their own complexity.

But we should also consider rounding up time or count limits to complete a chronological segment when one is started. Thus `Within a Day` might show 15 changes even if configured for a limit of 10. My own practice is never to set limits that don't align with whole days, weeks or months.

**Comment by opn** [comment]
Yes - omitting the helpful description and letting the author write it is best.

With regard to only providing variable date listings - we really do need more than that and have the ability to list a fixed number of changes regardless of the date. Sites can not have any changes for months or years - and yet you still want to see the last few changes - at the moment you can oly list all the changes which can take up a huge amount of variable screen real estate.

**Comment by paul90** [comment]
There appears to be two loosely related ideas presented. Below I add my thoughts on the suggestion of a new content verb `LAST`. I'm splitting the layout/presentation aspects into a separate issue (#30) so that they can receive the attention they deserve.

> I suggest a new verb "LAST n" (when n is an integer) which would display the last n changes.

Seems like a good idea, and easy to understand at a site level. So, it could be default turn neighbourhood off - in much the way a `ROSTER` does.

I assume that `LAST 3` with neighbours included only shows 3 pages (and not the last 3 pages updated on each site within the neighbourhood).

> Sites can not have any changes for months or years - and yet you still want to see the last few changes

I wonder... 

`SINCE` shows activity since a time relative to now. This is nice with an active site or neighbourhood. Bit... A for a currently inactive site it might be nice to have a modifier that changes that to show activity relative to the last time the site was active. So, for example, if the site has not been active since say 31 Jan 2016, asking for 1 weeks activity will show the activity during the last week in Jan 2016 rather than relative to now which would show nothing.

Not sure of the verb to use for this relative to last activity. For a neighbourhood I feel this would work relative to the last activity across the neighbourhood and not for each site individually.

**Comment by opn** [comment]
> I assume that LAST 3 with neighbours included only shows 3 pages (and not the last 3 pages updated on each site within the neighbourhood).

Yes - it should only show the last 3 pages in total.

> SINCE shows activity since a time relative to now. This is nice with an active site or neighbourhood. Bit... A for a currently inactive site it might be nice to have a modifier that changes that to show activity relative to the last time the site was active. So, for example, if the site has not been active since say 31 Jan 2016, asking for 1 weeks activity will show the activity during the last week in Jan 2016 rather than relative to now which would show nothing.
> 
> Not sure of the verb to use for this relative to last activity. For a neighbourhood I feel this would work relative to the last activity across the neighbourhood and not for each site individually.

Great idea.

`BEFORE somedate`

would allow this. "somedate" could be a standard format, or simply months and years like "Jan 2015"

Then:

```
BEFORE last
SINCE 1 week
```

Would give the last weeks changes.

```
BEFORE jan 2016
LAST 3
```

would give the last 3 changes of 2015?

**Comment by opn** [comment]
Also and not directly related to this issue - in a discussion with @WardCunningham with regard to the RSS plugin we suggested that it would be good to expose an RSS feed from the results of an Activity plugin. Or should that be simply the ROSTER?

This way we can create highly customised RSS feeds from a group of sites. The RSS feed would be created from the page title, the last modified date, and the first paragraph - all of which are available in the site map I believe?

I mention this now in case it makes sense to add the ability to the ACTIVITY plugin so that it can publish this information for other plugins - like the RSS plugin - or indeed in case it makes sense to roll up the ability to publish RSS directly into the ACTIVITY or ROSTER plugins so we no longer need a separate RSS plugin?
